quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMapper.java

```diff
@@ -96,6 +100,17 @@ ConfigValue getConfigValue(String name, ConfigSourceInterceptorContext context)
         // try to obtain the value for the property we want to map first
         ConfigValue config = convertValue(context.proceed(from));
 
+        if (config != null && getDeprecatedMetadata().isPresent()) {
+            String logMsg = String.format("Deprecated option in use: \"%s\" from %s.", option.getKey(), config.getConfigSourceName());
+            if (getDeprecatedMetadata().get().getNote() != null) {
+                logMsg += " " + getDeprecatedMetadata().get().getNote();
+            }
+            if (!getDeprecatedMetadata().get().getNewOptionsKeys().isEmpty()) {
+                logMsg += String.format(" Consider using: %s.", String.join(", ", getDeprecatedMetadata().get().getNewOptionsKeys()));
+            }
+            logger.warn(logMsg);
+        }
```

Comment:
It'd be probably good to store some context about these deprecations and at the end of processing these opts just print it to a console?

Reply:
Similarly as we do for errors, right? I think it will bring more consistency, yeah ...

Perhaps as clear entry like:

```
You are using deprecated options:

* option-foo: <note>
* option-bar: <note>

These options will be removed in the feature and no longer available from future release.
```

Something better than this ...

Reply:
If we want to be really strict about it, we could even fail the build/start commands and introduce a `--use-deprecated` option to make it really painful and force people to migrate. As well as force us to take real care when designing options and deprecating them :)

Reply:
Aggregating the log message to include all used deprecated options sounds good. Next week I'll try to figure out the way to do it.

Reply:
Reworked that so the deprecation log message is not printed only once and aggregates all deprecated options in use.

I would not be too strict (require the users to explicitly set `--use-deprecated true`) as deprecating something should not itself be a breaking change requiring you to set a new switch.